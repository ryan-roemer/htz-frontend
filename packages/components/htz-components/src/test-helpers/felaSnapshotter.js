import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
