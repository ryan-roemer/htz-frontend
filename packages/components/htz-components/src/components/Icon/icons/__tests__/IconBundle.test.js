
/* global window */
/* *************************************************************** *
 * THIS IS AN AUTO GENERATED FILE. PLEASE DO NOT EDIT IT DIRECTLY.
 *
 * If you want to change the styleguide example, it is generated
 * from the `embedsFileTemplate.js` file is this directory.
 * *************************************************************** */
import React from 'react';
import felaSnapshotter from '../../../../test-helpers/felaSnapshotter';
import IconBundle from '../IconBundle';

describe('<IconBundle />', () => {
  /* The error tests are left as documentation, but are commented out for the redability of the test results */
  // describe('Errors', () => {
  //   it('throw an error when the value passed to the "size" prop isn't a number', () => {
  //     expect(() => felaSnapshotter(<IconBundle size="2rem" />)).toThrow(
  //       /^An Icon's "size" prop may only be passed a "number"/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconBundle color="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconBundle color={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconBundle fill="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconBundle fill={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  // });
  describe('DOM element', () => {
    testCase('render correctly when no props are passed', <IconBundle />);
    testCase(
      'pass attributes defined in the "attrs" prop to the DOM element',
      <IconBundle attrs={{ 'aria-hidden': true, }} />
    );
    testCase('add "id" attribute to DOM element', <IconBundle id="hasId" />);
    testCase(
      'add "onClick" attribute to DOM element',
      <IconBundle id={e => console.log('a')} />
    );
  });

  describe('styles', () => {
    testCase("set icon's color", <IconBundle color={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's color responsively",
      <IconBundle
        color={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's fill", <IconBundle fill={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's fill responsively",
      <IconBundle
        fill={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's size", <IconBundle size={6} />);
    testCase(
      "set icon's size responsively",
      <IconBundle
        size={[ { until: 'xl', value: 6, }, { from: 'xl', value: 7, }, ]}
      />
    );
    testCase(
      'set misc styles on an icon',
      <IconBundle miscStyles={{ position: 'relative', }} />
    );
    testCase(
      'responsively set misc styles on an icon',
      <IconBundle
        miscStyles={{
          position: [
            { until: 'xl', value: 'relative', },
            { from: 'xl', value: 'static', },
          ],
        }}
      />
    );
  });
});

function testCase(name, Component, { only = false, } = {}) {
  const testFunc = only ? it.only : it;
  testFunc(name, () => {
    const { component, styles, } = felaSnapshotter(Component);
    expect(component).toMatchSnapshot();
    expect(styles).toMatchSnapshot();
  });
}
