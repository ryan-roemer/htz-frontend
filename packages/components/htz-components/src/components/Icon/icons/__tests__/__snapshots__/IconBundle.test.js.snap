// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconBundle /> DOM element add "id" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id="hasId"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> DOM element add "id" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconBundle /> DOM element add "onClick" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id={[Function]}
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> DOM element add "onClick" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconBundle /> DOM element pass attributes defined in the "attrs" prop to the DOM element 1`] = `
<div>
  <svg
    aria-hidden={true}
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> DOM element pass attributes defined in the "attrs" prop to the DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconBundle /> DOM element render correctly when no props are passed 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> DOM element render correctly when no props are passed 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconBundle /> styles responsively set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> styles responsively set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{position:relative}}@media (min-width: 80em){.htz-e{position:static}}"`;

exports[`<IconBundle /> styles set icon's color 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> styles set icon's color 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:#0B7EB5}.htz-c{fill:#FFF}"`;

exports[`<IconBundle /> styles set icon's color responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> styles set icon's color responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-d{fill:#FFF}@media (max-width: 79.9375em){.htz-b{color:#0B7EB5}}@media (min-width: 80em){.htz-c{color:#2D2D2D}}"`;

exports[`<IconBundle /> styles set icon's fill 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> styles set icon's fill 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#0B7EB5}"`;

exports[`<IconBundle /> styles set icon's fill responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> styles set icon's fill responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}@media (max-width: 79.9375em){.htz-c{fill:#0B7EB5}}@media (min-width: 80em){.htz-d{fill:#2D2D2D}}"`;

exports[`<IconBundle /> styles set icon's size 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> styles set icon's size 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{font-size:6rem}"`;

exports[`<IconBundle /> styles set icon's size responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> styles set icon's size responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 79.9375em){.htz-d{font-size:6rem}}@media (min-width: 80em){.htz-e{font-size:7rem}}"`;

exports[`<IconBundle /> styles set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
     
    <path
      d="M162 170H22.4c-.2 0-.4-.2-.4-.4V43.4c0-.2.2-.4.4-.4h178.2c.2 0 .4.2.4.4V84h-6V49H28v115h134v6zm0 9H21c-3.3 0-6-2.7-6-6V42c0-3.3 2.7-6 6-6h182c3.3 0 6 2.7 6 6v42h6V42c0-6.6-5.4-12-12-12H21c-6.6 0-12 5.4-12 12v131c0 6.6 5.4 12 12 12h141v-6zm74-82h-56c-3.3 0-6 2.7-6 6v119c0 3.3 2.7 6 6 6h56c3.3 0 6-2.7 6-6V103c0-3.3-2.7-6-6-6m0-6c6.6 0 12 5.4 12 12v119c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12V103c0-6.6 5.4-12 12-12h56zm-3 22h-49v97h49v-97m5.6-6c.2 0 .4.2.4.4v108.2c0 .2-.2.4-.4.4h-60.2c-.2 0-.4-.2-.4-.4V107.4c0-.2.2-.4.4-.4h60.2zM218 222c0-1.1-.9-2-2-2h-13c-1.1 0-2 .9-2 2s.9 2 2 2h13c1.1 0 2-.9 2-2zm-7-120v-1c0-1.1-.9-2-2-2h-1c-1.1 0-2 .9-2 2v1c0 1.1.9 2 2 2h1c1.1 0 2-.9 2-2zm13.3 19.1l-33.2 33.2 3.5 3.5 33.2-33.2-3.5-3.5zm-12.1 33.1l-21.1 21.1 3.5 3.5 21.1-21.1-3.5-3.5zM163 210c0-8.3-6.5-15-14.4-15H76.4c-7.9 0-14.4 6.7-14.4 15v6h6v-6c0-5 3.8-9 8.4-9h72.2c4.6 0 8.4 4 8.4 9v6h6v-6zm-73-25h-5v11h5v-11zm49 0h-5v11h5v-11z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconBundle /> styles set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{position:relative}"`;
